src/app/model/webapi/knora/v2/convert-jsonld.ts

```diff
@@ -164,7 +164,10 @@ export module ConvertJSONLD {
 
             case AppConfig.DecimalValue:
 
-                let decimalValue = new ReadDecimalValue(propValue['@id'], propIri, propValue[AppConfig.decimalValueAsDecimal]);
+                // a decimal value is represented as a string in order to preserve its precision
+                const decVal: number = parseFloat(propValue[AppConfig.decimalValueAsDecimal]);
```

Comment:
I noticed that decimal values are represented as JSON strings because of the precision.

Reply:
Yes. I think actually we should represent them like this:

```jsonld
{
    "knora-api:decimalValueHasDecimal": {
        "@type": "xsd:decimal",
        "@value": "1000000.00000001"
    }
}
```

Otherwise I think conversion to Turtle won't work.

Reply:
But in that case it would have to  be done consistently for all types?

Reply:
At least for some types. You wouldn't need it for integers or booleans, because their types are clear from the syntax of JSON, and [RDF4J Rio](http://docs.rdf4j.org/programming/#_parsing_and_writing_rdf_with_rio) converts JSON integers and booleans to Turtle integers and booleans. But definitely a custom datatype like `knora-api:Date` or `knora-api:Color` (in the simple schema) would need a `@type` in JSON-LD, otherwise it would just look like a string to Rio. I can do this in PR #851, as well as dealing with issue #854.

Reply:
> When do you plan to do this?

I'd like to finish https://github.com/dhlab-basel/Knora/pull/851 next week, so https://github.com/dhlab-basel/Knora/issues/854 can be resolved as soon as possible, because it affects the ontology editor work being done for SALSAH 2, as well as export from the old SALSAH.

Reply:
Ok, since you are here in Basel next week, we can look at it.